src/odemis/gui/comp/overlay/shapes.py

```diff
@@ -182,6 +184,20 @@ def _activate_shapes(self, flag=True):
         for shape in self._shapes:
             shape.active.value = flag
 
+    def _deselected_shapes(self):
+        """Deselect shapes except the selected shape."""
+        for shape in self._shapes:
+            if shape.selected.value and shape != self._selected_shape:
+                shape.selected.value = False
+
+    def _unbind_shape_events(self, shape: EditableShape):
+        """Unbind shape's events bound to the canvas."""
+        shape.cnvs.Unbind(wx.EVT_LEFT_DOWN, handler=shape.on_left_down)
+        shape.cnvs.Unbind(wx.EVT_LEFT_UP, handler=shape.on_left_up)
+        shape.cnvs.Unbind(wx.EVT_RIGHT_DOWN, handler=shape.on_right_down)
+        shape.cnvs.Unbind(wx.EVT_RIGHT_UP, handler=shape.on_right_up)
+        shape.cnvs.Unbind(wx.EVT_MOTION, handler=shape.on_motion)
```

Comment:
This works... but it just adds complexity on top of complexity. You're essentially undoing the Overlay code. I'd suggest to just not consider the shapes as Overlays. Their .draw() function can be called from the ShapeOverlay.draw(). They don't need to be part of the canvas's world overlays. 
You can merge RectanglePointsSelectOverlay into RectangleOverlay (because it's the only class that uses it). Then, pretty much the only thing that is left from the "WorldOverlay" class is the `.active` VA, which you can duplicate... and probably it's not even necessary, but can just be passed as an argument to .draw() (by the ShapeOverlay).

Reply:
Do you foresee  in the future that someone will use a (+) button to draw a single `RectanglePointsSelectOverlay`, `PolygonOverlay`, `EllipseOverlay`, or `RectangleOverlay` @pieleric ? I still left those classes inherit the `WorldOverlay` because of it

Reply:
No, I don't think this will be a coming feature. Especially, even if it's a case, that can still be implemented by having a ShapeOverlay + a way to tell it to create a shape. (Like this is done for the GadgetOverlay.